bazel/envoy_library.bzl

```diff
@@ -219,6 +219,50 @@ def envoy_cc_posix_library(name, srcs = [], hdrs = [], **kargs):
         **kargs
     )
 
+# Used to specify a library that only builds on POSIX excluding Linux
+def envoy_cc_posix_without_linux_library(name, srcs = [], hdrs = [], **kargs):
+    envoy_cc_library(
+        name = name + "_posix",
+        srcs = select({
+            "@envoy//bazel:windows_x86_64": [],
+            "@envoy//bazel:linux_x86_64": [],
+            "@envoy//bazel:linux_aarch64": [],
+            "@envoy//bazel:linux_ppc": [],
+            "@envoy//bazel:linux_mips64": [],
```

Comment:
We should refactor this a bit with `config_settings_group`: https://github.com/bazelbuild/bazel-skylib/blob/master/docs/selects_doc.md

Reply:
actually, I noticed that on some places (e.g.: for quiche) we just match on:

```
"@envoy//bazel:linux"
```

whereas in most other cases we explicitly enumerate each Linux arch, when is one way preferred over the other?

Reply:
If that works go for it. Sometime we have to differentiate architecture but this is not the case.

Reply:
Yeah I think it does, I had it that way initially and then I switched over when debugging something. Let me try.